zebra-rpc/src/methods.rs

```diff
@@ -750,9 +750,16 @@ where
 
         let mut state = self.state.clone();
         let verbosity = verbosity.unwrap_or(DEFAULT_GETBLOCK_VERBOSITY);
-        let self_clone = self.clone();
-
+        let network = self.network.clone();
         let original_hash_or_height = hash_or_height.clone();
+
+        // If verbosity requires a call to `get_block_header`, resolve it here
+        let get_block_header_future = if matches!(verbosity, 1 | 2) {
+            Some(self.get_block_header(original_hash_or_height.clone(), Some(true)))
+        } else {
+            None
+        };
```

Comment:
Optional/Nitpick/Subjective: This may actually be a little less readable, but it's tempting to use `then()` here.

```suggestion
        let get_block_header_future = matches!(verbosity, 1 | 2)
            .then(|| self.get_block_header(original_hash_or_height.clone(), Some(true)));
```